lib/inference_engines/runner/step_execution.rb

```diff
@@ -30,49 +30,53 @@ def debug_log(params)
         end
```

Comment:
The changes here are about branching the runner between two modes of operation. With hindsight these may have been better hooked into different runners entirely.

Essentially though this avoids the need to spin up a separate process, educing time, memory, and avoiding a whole round of serialization, deserialization and database loading.